hooks/src/index.js

```diff
@@ -27,22 +27,6 @@ const RAF_TIMEOUT = 100;
 let prevRaf;
 
 options._diff = vnode => {
-	if (
-		typeof vnode.type === 'function' &&
```

Comment:
I guess the current solution is aimed at making stuff more efficient i.e. only moving to call-site we could reduce the byte impact by just removing this condition so it applies to both DOM and fn nodes

Reply:
Turns out we can't rely on the tree position alone. When two different components are rendered conditionally and both call `useId` we'd return the same id if we base our logic on the tree position. See this comment: https://github.com/preactjs/preact/issues/3772#issuecomment-1290393609

Reply:
Yea, that's all right as they are switched out so no dupes in dom